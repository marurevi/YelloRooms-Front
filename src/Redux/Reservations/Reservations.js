import { sendGet, sendDelete } from '../../API/api';

const GET_RESERVATIONS = 'reservations/reservations/SET_RESERVATIONS';
const DELETE_RESERVATION = 'reservations/reservations/DELETE_RESERVATION';
export const getReservations = () => async (dispatch) => {
  const reservations = await sendGet('reservations').catch((error) => {
    console.error(error); // eslint-disable-line no-console
  });
  dispatch({ type: GET_RESERVATIONS, payload: reservations.data });
};

export const deleteReservation = (id) => async (dispatch) => {
  sendDelete(`reservations/${id}`)
    .then(() => {
      dispatch({ type: DELETE_RESERVATION, payload: id });
    })
    .catch((error) => {
      console.log(error); // eslint-disable-line no-console
    });
};

const reservationsReducer = (state = [], action) => {
  switch (action.type) {
    case GET_RESERVATIONS:
      return action.payload;
    case DELETE_RESERVATION:
      return [...state].filter(
        (reservation) => reservation.id !== action.payload,
      );
    default:
      return state;
  }
};

export default reservationsReducer;
