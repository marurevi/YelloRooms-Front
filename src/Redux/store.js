import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
} from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import roomsReducer from './Rooms/rooms';
import reservationsReducer from './Reservations/Reservations';
import userReducer from './Authenticate/authentication';

const reducers = combineReducers({
  rooms: roomsReducer,
  reservations: reservationsReducer,
  user: userReducer,
});

const store = createStore(reducers, applyMiddleware(thunk, logger));
export default store;
