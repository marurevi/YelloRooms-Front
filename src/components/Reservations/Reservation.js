import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import DeleteReservation from '../partials/DeleteReservation';

const Reservation = ({ reservationId }) => {
  const reservation = useSelector((store) => store.reservations.find(
    (reservation) => reservation.id === reservationId,
  ));
  const reservedRoom = useSelector((store) => store.rooms.rooms.find(
    (room) => Number(room.id) === reservation.attributes.room_id,
  ));
  return (
    <div
      className="reservation"
      style={{ backgroundImage: `url(${reservedRoom.attributes.photo})` }}
    >
      <div className="container">
        <p>
          <span className="label">Reserved Room Name: </span>
          {reservedRoom.attributes.name}
        </p>
        <p>
          <span className="label">Reservation city: </span>
          {reservation.attributes.city}
        </p>
        <p>
          <span className="label">Reservation start date: </span>
          {reservation.attributes.start_date}
        </p>
        <p>
          <span className="label">Reservation end date: </span>
          {reservation.attributes.end_date}
        </p>
        <DeleteReservation id={reservation.id} />
      </div>
    </div>
  );
};

export default Reservation;

Reservation.propTypes = {
  reservationId: PropTypes.string.isRequired,
};
