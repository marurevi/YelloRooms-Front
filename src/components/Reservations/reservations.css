.reservation {
  border: 1px solid black;
  border-radius: 10px;
  padding: 5px;
  box-shadow: 0 0 10px silver;
  font-size: 1.2rem;
  color: darkred;
  max-width: max-content;
  display: inline-block;
  background-size: cover;
  margin: 15px;
}

.label {
  font-size: 1.5em;
  margin: 13px;
  text-transform: capitalize;
  line-height: 1.5em;
  color: #ffde59;
  text-shadow: black 2px 2px 2px;
}

.container {
  border-radius: 5px;
  padding: 5px;
  background-color: silver;
  color: black;
  font-weight: bolder;
  opacity: 0.9;
}

.container button {
  padding: 5px;
  color: white;
  background-color: #e11010;
}
