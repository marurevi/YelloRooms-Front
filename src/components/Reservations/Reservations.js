import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Reservation from './Reservation';
import { getReservations } from '../../Redux/Reservations/Reservations';
import './reservations.css';

const Reservations = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getReservations());
  }, []); // eslint-disable-line react-hooks/exhaustive-deps
  const userReservations = useSelector((state) => state.reservations.filter(
    (reservation) => reservation.attributes.user_id === state.user.id,
  ));
  return userReservations.length ? (
    <div>
      <h2>MY RESERVATIONS</h2>
      {userReservations.map((reservation) => (
        <Reservation reservationId={reservation.id} key={reservation.id} />
      ))}
    </div>
  ) : (
    <h2>There are not any reservations</h2>
  );
};

export default Reservations;
