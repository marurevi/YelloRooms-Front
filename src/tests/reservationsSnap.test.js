import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Reservations from '../components/Reservations/Reservations';
import store from '../Redux/store';

test('Reservations snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Reservations />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
