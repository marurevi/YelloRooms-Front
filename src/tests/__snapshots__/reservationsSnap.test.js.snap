// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservations snapshot 1`] = `
<h2>
  There are not any reservations
</h2>
`;
